cli/commons/src/Microsoft.Kiota.Cli.Commons/IO/JsonOutputFormatter.cs

```diff
@@ -0,0 +1,127 @@
+using System.Text.Json;
+using Spectre.Console;
+
+namespace Microsoft.Kiota.Cli.Commons.IO;
+
+/// <summary>
+/// The JSON output formatter
+/// </summary>
+public class JsonOutputFormatter : IOutputFormatter
+{
+    private readonly IAnsiConsole _ansiConsole;
+
+    /// <summary>
+    /// Creates a new JSON output formatter with a default console
+    /// </summary>
+    public JsonOutputFormatter() : this(AnsiConsole.Console)
+    {
+    }
+
+    /// <summary>
+    /// Creates a new JSON output formatter with the provided console
+    /// </summary>
+    /// <param name="console">The console to use</param>
+    public JsonOutputFormatter(IAnsiConsole console)
+    {
+        this._ansiConsole = console;
+    }
+
+    /// <inheritdoc />
+    public void WriteOutput(string content, IOutputFormatterOptions options)
+    {
+        if (options is JsonOutputFormatterOptions jsonOptions && jsonOptions.OutputIndented)
+        {
+            var result = ProcessJson(content, jsonOptions.OutputIndented);
+            _ansiConsole.WriteLine(result);
+        }
+        else
+        {
+            _ansiConsole.WriteLine(content);
+        }
+    }
+
+    /// <inheritdoc />
+    public void WriteOutput(Stream content, IOutputFormatterOptions options)
+    {
+        using var reader = new StreamReader(content);
+        var strContent = reader.ReadToEnd();
+        if (options is JsonOutputFormatterOptions jsonOptions && jsonOptions.OutputIndented)
+        {
+            var result = ProcessJson(strContent, jsonOptions.OutputIndented);
+            _ansiConsole.WriteLine(result);
+        }
+        else
+        {
+            _ansiConsole.WriteLine(strContent);
+        }
+    }
+
+    /// <inheritdoc />
+    public async Task WriteOutputAsync(Stream content, IOutputFormatterOptions options, CancellationToken cancellationToken = default)
+    {
+        string resultStr;
+
+        if (options is JsonOutputFormatterOptions jsonOptions && jsonOptions.OutputIndented)
+        {
+            using var result = await ProcessJsonAsync(content, jsonOptions.OutputIndented, cancellationToken);
+            using var r = new StreamReader(result);
+            resultStr = await r.ReadToEndAsync();
+        }
+        else
+        {
+            using var reader = new StreamReader(content);
+            resultStr = await reader.ReadToEndAsync();
+        }
+
+        _ansiConsole.WriteLine(resultStr);
+    }
+
+    /// <summary>
+    /// Given a JSON input string, returns a processed JSON string with optional indentation
+    /// </summary>
+    /// <param name="input">JSON input string</param>
+    /// <param name="indent">Whether to return indented output</param>
+    private static string ProcessJson(string input, bool indent = true)
+    {
+        var result = input;
+        try
+        {
+            var jsonDoc = JsonDocument.Parse(input);
+            result = JsonSerializer.Serialize(jsonDoc, options: new() { WriteIndented = indent });
```

Comment:
Maybe we could probably create a local instance of `JsonSerializerOptions` to avoid having to create a new instance every time we call `JsonSerializer.Serialize` in the class since all the methods in the class use the same version of the instance.